### PR TITLE
Unit-testing Group View Composer

### PR DESCRIPTION
**Pull request by deleugpn**
```
# ./vendor/bin/phpunit
PHPUnit 6.2.1 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.0.16 with Xdebug 2.5.0
Configuration: /var/www/forks/laravel-forum/phpunit.xml

.....                                                               5 / 5 (100%)

Time: 16.24 seconds, Memory: 20.00MB

OK (5 tests, 15 assertions)
```

- Setup two separate Test Suite (Unit and Integration);
- Changed ViewComposer to rely on Interface rather than implementation to allow Mockery;
- Removed Integration Tests because `factory()` only works with complete database setup;
- Added Test for Group ViewComposer;
- Implemented GrahamCampbell solution to count mockery towards assertion on `TestCase::tearDown` (https://github.com/mockery/mockery/issues/376#issuecomment-56095477)

### CODE REVIEW

File: tests/Unit/GroupComposerTest.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+class GroupComposerTest extends TestCase
+{
+    public function testViewHasAllGroups()
+    {
+        factory(Group::class)->times(rand(0, 10))->create();
+        $view = Mockery::mock(View::class);
+        $view->shouldReceive('with')->withArgs(['groups', Mockery::type(Collection::class)])->andReturnSelf()->once();
+
+        $viewComposer = new GroupComposer();
+        $viewComposer->compose($view);
```
Comment (Dylan-DPC) [inline tests/Unit/GroupComposerTest.php]: not asserting anything ... ?

Comment (deleugpn) [inline tests/Unit/GroupComposerTest.php]: The mock is the assertion. If the method `with` doesn't get called with the given attributes, it fails.

Comment (Dylan-DPC) [inline tests/Unit/GroupComposerTest.php]: how?

Comment (deleugpn) [inline tests/Unit/GroupComposerTest.php]: I don't really know how that happens under the hood. All I know is that if you don't call the method, the test fails.

**Comment by deleugpn** [comment]
As explained on Slack: 

basically it goes like this: You setup the mock at a specific conditions
- If the code tries to call something that the mock is not prepared to handle, it throws exception;
- If the code doesn't call something that the mock is expecting, it throws an exception;

if no exception is thrown, you assume it passed; It's an assertion by exclusion

**Pull request merged**